scipy/stats/_continuous_distns.py

```diff
@@ -6898,6 +6910,35 @@ def _ppf(self, q, c, d):
                       1 - np.sqrt((1 - q) * (d - c + 1) * (1 - d))]
         return np.select(condlist, choicelist)
 
+    def _munp(self, n, c, d):
+        # Using van Dorn parameterization as given on wikipedia, with
+        # a=bottom left, b=top left, c=top right, d=bottom right, then
+        # E[X^(k+2)] = h/(k(k-1)) [(d^k-c^k)/(d-c) - ((b^k - a^k)/(b-a)]
+        # with h = 2/((d+c) - (a+b)).  The corresponding parameterization
+        # in scipy, has a'=loc, b'=loc+c*scale, c'=loc+d*scale, d'=loc+scale,
+        # which for standard form reduces to a'=0, b'=c, c'=d, d'=1.
+        # Simplifying E[X^n] gives the dc' term as (d^(n+2) - 1)/(d - 1) and
+        # the ab' term as c^(n-1). The dc' term has numerical difficulties
+        # near d=1, so replace (1 - c^k)/(1-c) with expm1(k*log(c))/(c-1).
```

Comment:
Sorry, I'm not going through the math, just the comment. Where does `(1 - c^k)/(1-c)` come from if the `dc'` term is `(d^(n+2) - 1)/(d - 1)`? And is there a reference for this substitution being more stable?

Reply:
I'll drop references to `k+2 = n` since it is both (a) confusing, and (b) wrong [I should have written `k-2 = n`].  Also, I'll switch to the Kracker  et al. form which uses `(a, c, d, b)` instead of the van Dorn `(a, b, c, d)` for the points on the trapezoid since this avoids the `(b, c) <=> (c, d)` confusion when translating the equations to the scipy standard form. 

Re: stability, the [numpy docs](https://docs.scipy.org/doc/numpy/reference/generated/numpy.expm1.html) simply say that `expm1` is more precise.  You can see this from an expansion of the Taylor series about `x=0`, which for `e^x-1` is `∑_{k=1}^∞ x^k/k!`.  Since the series doesn't start at k=0 you don't get the cancellation from subtraction that you do with a simple evaluation of `e^x-1`. I checked against 500-bit arithmetic using `mpmath` by temporarily modifying the code in [sasmodels](https://github.com/SasView/sasmodels/blob/master/explore/precision.py) to include various forms of the expression.  I can include the plots if you are interested, but the result is that it generally within 1 ulp but occasionally goes very slightly above..